pkg/controllers/webhookcerts/cert_requester.go

```diff
@@ -0,0 +1,380 @@
+// it's requesting token reviews from the oauth-apiserver
+func NewWebhookAuthenticatorCertRequester(
+	operatorClient v1helpers.OperatorClient,
+	csrClient certv1client.CertificateSigningRequestInterface,
+	csrInformers certv1informers.CertificateSigningRequestInformer,
+	secretClient corev1client.SecretInterface,
+	secretInformers corev1informers.SecretInformer,
+	eventsRecorder events.Recorder,
+) factory.Controller {
+	c := &webhookAuthenticatorCertRequester{
+		operatorClient: operatorClient,
+
+		csrClient: csrClient,
+		csrLister: csrInformers.Lister(),
+
+		secretClient: secretClient,
+		secretLister: secretInformers.Lister(),
+	}
+
+	return factory.New().
+		WithSync(c.syncWithConditions).
+		WithSyncDegradedOnError(operatorClient).
+		WithInformers(operatorClient.Informer()).
+		WithFilteredEventsInformers(nameFilter(csrName), csrInformers.Informer()).
+		WithFilteredEventsInformers(nameFilter(certSecretName), secretInformers.Informer()).
+		ToController("WebhookAuthenticatorCertRequester", eventsRecorder.WithComponentSuffix("webhook-authenticator-cert-requester"))
+}
+
+func (c *webhookAuthenticatorCertRequester) syncWithConditions(ctx context.Context, syncCtx factory.SyncContext) error {
+	progressing, err := c.sync(ctx, syncCtx)
+
+	if progressing != nil {
+		var progressingText string
+		progressingStatus := operatorv1.ConditionFalse
+		progressingReason := "CertificateAvailable"
+		if *progressing {
+			progressingText = "waiting for CSR to be signed"
+			progressingStatus = operatorv1.ConditionTrue
+			progressingReason = "WaitingForCertificate"
+		}
+
+		_, _, statusErr := v1helpers.UpdateStatus(c.operatorClient,
+			v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+				Type:    "AuthenticatorCSRProgressing",
+				Status:  progressingStatus,
+				Reason:  progressingReason,
+				Message: progressingText,
+			}))
+
+		if statusErr != nil {
+			if err != nil {
+				klog.Error("failed to update operator status: %v", err)
+				return err
+			}
+			return statusErr
+		}
+	}
+
+	return err
+}
+
+func (c *webhookAuthenticatorCertRequester) sync(ctx context.Context, syncCtx factory.SyncContext) (progressing *bool, err error) {
+	key, cert, keyRotated, err := c.getKeyCertFromSecret(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if keyRotated {
+		if err := c.resubmitCSR(ctx, key); err != nil {
+			return nil, err
+		}
+		return pbool(true), nil
+	}
+
+	if cert != nil {
+		// check that the certificate is not expiring
+		certValidityDuration := cert.NotAfter.Sub(cert.NotBefore)
+		if time.Now().Before(cert.NotBefore.Add(time.Duration(0.8 * float64(certValidityDuration)))) {
+			return pbool(false), nil
+		}
+
+		klog.V(5).Info("the authenticator's certificate is almost expired, will schedule a new CSR creation")
+
+		// delete the cert/key secret and schedule a new sync
+		if err := c.secretClient.Delete(ctx, certSecretName, metav1.DeleteOptions{}); err != nil {
```

Comment:
simple update would do as well

Reply:
the update must delete all the keys in the secret's `data`, deleting seems simpler

Reply:
`deleting` means that we would issue at least `2` requests, one for delete and the other one for create
`updating` would mean a single request - we could delete the old data and regen the new key.